src/components/AddUserForm/index.tsx

```diff
@@ -11,110 +14,154 @@ import {
   SelectUserRole,
 } from './style';
 
+interface UserCreationErrorData {
+  birthDateError: boolean;
+  emailError: boolean;
+  phoneError: boolean;
+  passwordError: boolean;
+}
+
 enum Role {
   USER = 'user',
   ADMIN = 'admin',
 }
 
-type RoleType = typeof Role[keyof typeof Role];
+export type RoleType = typeof Role[keyof typeof Role];
 
 export const AddUserForm: React.FC = () => {
-  const [email, setEmail] = useState('');
   const [name, setName] = useState('');
-  const [phone, setPhone] = useState('');
+  const [email, setEmail] = useState('');
   const [birthDate, setBirthDate] = useState('');
-  const [role, setRole] = useState<RoleType>('' as RoleType);
+  const [phone, setPhone] = useState('');
+  const [role, setRole] = useState<RoleType>();
+  const [password, setPassword] = useState('');
 
-  const [userCreationError, setUserCreationError] = useState(false);
+  const [createUser] = useMutation(ADD_USER);
+  const [userCreationError, setUserCreationError] = useState({} as UserCreationErrorData);
```

Comment:
aqui acho que é melhor passar o tipo para o hook
```suggestion
  const [userCreationError, setUserCreationError] = useState<UserCreationErrorData>({});
```